### DISCUSSION TITLE
Geometry transform : inplace version?

### DISCUSSION THREAD

**Discussion by ghost** [boostorg/geometry#1090] [Q&A] [answered]
Hi,
Could the boost::geometry::strategy::transform::rotate/translate/scale be made in place?

It seems that having a different output geometry is a waste. Since we don't modify the number of points, it should be doable.

What is your opinion?
Thanks

**Comment by awulkiew** [comment, marked as answer]
I can think of two reasons why the interface looks like this:
- it is based on STL and `std::transform` takes two ranges,
- `geometry::transform` returns `bool` indicating whether or not transformation was successful. So theoretically it is possible that transform might fail and in this case a user could end up with a geometry that was partially modified. With that said AFAIK transformations that are supported by default never fail.

AFAIU to meet your expectations it would probably be enough to make sure that the algorithm works for the same geometry passed as input and output, correct?

For now:
```
bg::for_each_point(geometry, [](auto &pt) {
    auto tmp = pt;
    bg::transform(tmp, pt, per_point_strategy);
});
```
It's possible that `tmp` is not needed, it depends on the strategy.

**Comment by ghost** [comment]
That's interesting and it's the way I will do it now.
Thanks!
